### PR TITLE
Make sure all reaction names are recognized as valid tally scores

### PR DESCRIPTION
**Pull request by paulromano**
@makeclean noted that some reaction names are not recognized as valid tally scores. For example, if you try using `(n,5n)` in a tally, OpenMC tells you that it is an invalid score, even though there is a corresponding ENDF reaction (MT=152). This PR refactors how we check for tally scores; importantly, it establishes a name: MT mapping based on the existing MT: name mapping, so they should always be consistent. Should we need to add additional reactions in the future, there are now fewer places the code needs to be modified.

### CODE REVIEW

File: src/reaction.cpp [hunk 1]
```diff
@@ -174,32 +223,98 @@ const std::unordered_map<int, std::string> REACTION_NAME_MAP {
...
+    if (level >= 1 && level <= 40) {
+      REACTION_NAME_MAP[50 + level] = fmt::format("(n,n{})", level);
+    }
+    REACTION_NAME_MAP[600 + level] = fmt::format("(n,p{})", level);
+    REACTION_NAME_MAP[650 + level] = fmt::format("(n,d{})", level);
+    REACTION_NAME_MAP[700 + level] = fmt::format("(n,t{})", level);
+    REACTION_NAME_MAP[750 + level] = fmt::format("(n,3He{})", level);
+    REACTION_NAME_MAP[800 + level] = fmt::format("(n,a{})", level);
+    if (level <= 15) {
+      REACTION_NAME_MAP[875 + level] = fmt::format("(n,a{})", level);
```
Comment (GiudGiud) [inline src/reaction.cpp:241]: the serpent wiki page has these as n,2n reactions. I cant find them on the lanl page I use usually. Can you please send me a link to what you used?

File: src/reaction.cpp [hunk 2]
```diff
@@ -174,32 +223,98 @@ const std::unordered_map<int, std::string> REACTION_NAME_MAP {
...
+  if (REACTION_TYPE_MAP.empty()) initialize_maps();
+
+  // Check if type map has an entry for this reaction name
+  auto it = REACTION_TYPE_MAP.find(name);
+  if (it != REACTION_TYPE_MAP.end()) {
+    return it->second;
+  }
+
+  // Alternate names for several reactions
+  if (name == "(n,total)") {
```
Comment (GiudGiud) [inline src/reaction.cpp:283]: If these names are in reaction type map wont this routine have returned already?

File: src/reaction.cpp (hunk 1 above)
Comment (paulromano) [reply src/reaction.cpp:241]: Yes, these should be n,2n reactions. Thanks for catching this! The most up-to-date source is the [ENDF-6 formats manual](https://www.osti.gov/biblio/1425114-endf-formats-manual-data-formats-procedures-evaluated-nuclear-data-files).

File: src/reaction.cpp (hunk 2 above)
Comment (paulromano) [reply src/reaction.cpp:283]: It looks like (n,total) needs to be handled separately actually. All the other names are alternate names that don't appear in the map, but (n,total) does appear, so I'm going to change this to check for it before the map.

**Pull request merged**